aea/manager/project.py

```diff
@@ -301,7 +301,8 @@ def set_overrides(
                 )
 
         overrides["component_configurations"] = component_configurations
-        return self.agent_config_manager.update_config(overrides)
+        self.agent_config_manager.update_config(overrides)
+        self._ensure_private_keys()
```

Comment:
Not sure if this makes sense here; I'd prefer it to be only called if agent_overrides are not None; to make explicit fact that it depends on changes in agent overrides